packages/autocomplete/src/elements/Autocomplete.js

```diff
@@ -0,0 +1,270 @@
+  };
+
+  renderMenuItems = getItemProps => {
+    const { noOptionsMessage, renderOption, selectedValue } = this.props;
+    const { focusedKey } = this.state;
+
+    const matchingOptions = this.getMatchingOptions().map(option => {
+      const checked = selectedValue === option.value;
+
+      const props = getItemProps({
+        key: option.value,
+        focused: focusedKey === option.value,
+        checked,
+        'aria-selected': checked
+      });
+
+      props.children = option.label;
+
+      if (renderOption) {
+        return renderOption(props);
+      }
+
+      return <Item {...props} />;
+    });
+
+    if (matchingOptions.length > 0) {
+      return matchingOptions;
+    }
+
+    return <StyledNoItemsMessage>{noOptionsMessage}</StyledNoItemsMessage>;
+  };
+
+  render() {
+    const {
+      label,
+      'aria-label': ariaLabel,
+      hint,
+      disabled,
+      options,
+      selectedValue,
+      onChange,
+      placeholder,
+      maxHeight,
+      renderDropdown,
+      inputRef,
+      message,
+      small,
+      validation
+    } = this.props;
+    const { isOpen, focusedKey, isFocused, inputValue } = this.state;
+
+    const optionDictionary = options.reduce((dictionary, option) => {
+      dictionary[option.value] = option.label;
+
+      return dictionary;
+    }, {});
+
+    return (
+      <FieldContainer>
+        {({ getLabelProps, getInputProps: getFieldInputProps, getHintProps }) => (
+          <TextGroup>
+            {label && <Label {...getLabelProps()}>{label}</Label>}
+            {hint && <Hint {...getHintProps()}>{hint}</Hint>}
+            <AutocompleteContainer
+              isOpen={isOpen}
+              focusedKey={focusedKey}
+              onSelect={selectedKey => {
+                onChange && onChange(selectedKey);
+              }}
+              onStateChange={newState => {
+                if (!newState.isOpen) {
+                  newState.inputValue = '';
+                }
+
+                this.setState(newState);
+              }}
+              trigger={({
+                getTriggerProps,
+                getInputProps,
+                triggerRef,
+                inputRef: triggerInputRef
+              }) => {
+                let triggerProps = getTriggerProps({
+                  open: isOpen,
+                  focused: isFocused || isOpen,
+                  select: true,
+                  small,
+                  validation,
+                  'aria-label': ariaLabel,
+                  inputRef: ref => {
+                    this.wrapperRef = ref;
+                    triggerRef(ref);
+                  }
+                });
+
+                if (disabled) {
+                  triggerProps = { disabled: true, small, validation };
+                }
+
+                return (
+                  <FauxInput {...triggerProps}>
+                    {!isOpen && <span>{optionDictionary[selectedValue]}</span>}
```

Comment:
The behavior of this span isn't working well.

* should be seeing an i-beam style `text` cursor across the entire width and height of the select (except for the dropdown cursor strike area)
* a click on the autocomplete should allow immediate typing (as it does with `examples/autocomplete.md`)
* need to figure out the 1px bump occurring on `isOpen`

The action should feel a little more like the **ContentEditable &lt;div&gt; Dropdown** on https://garden.zendesk.com/css-components/forms/dropdown/

Reply:
> should be seeing an i-beam style text cursor across the entire width and height of the select (except for the dropdown cursor strike area)

This is now styled correctly.

> a click on the autocomplete should allow immediate typing (as it does with examples/autocomplete.md)

This is working as expected for me. Can you help reproduce this?